app/controllers/search_controller.rb

```diff
@@ -56,11 +56,22 @@ def tags
   end
 
   def chat_channels
-    ccm_docs = Search::ChatChannelMembership.search_documents(
-      params: chat_channel_params.merge(user_id: current_user.id).to_h,
-    )
-
-    render json: { result: ccm_docs }
+    if chat_channel_params["discoverable"].present?
```

Comment:
All the chat channel docs should be able to be returned together using a single search without the need for an `if/else` statement. This endpoint should be able to remain exactly how it is with us just tweaking the params that we are sending to it. The params we send should include `user_id: [<the_user_id>, "all"]` then we run our search and get back all relevant docs. 

We don't want to make multiple calls to the server or Elasticsearch when we could do it with a single call and then let the frontend sort out the documents. That will greatly decrease the number of searches we have to run and will help keep our application running fast.

Reply:
@mstruve Just optimized the code further. 

The complexity of repetitive channels will be too much to handle in frontend. Also if in future let say there are 100+ user in a channel then it will be a bigger mess. But I agree that multiple calls to Elasticsearch also should be avoided. So the new code addressed both the things. Please have a look and if further optimization is required we are up for it. 🙂